step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -272,7 +272,21 @@ public String getDisplayName() {
      * A type of a parameter to a class.
      */
     public static abstract class ParameterType {
-        ParameterType() {}
+        @Nonnull
+        private final Type actualType;
+
+        public Type getActualType() {
+            return actualType;
+        }
+
+        public Class<?> getType() {
+            return (Class) getActualType();
```

Comment:
:bug: this should be pushed down into those subclasses which can actually _guarantee_ that the type is in fact a `Class`, and not something else.